src/RealtimeEffectPanel.cpp

```diff
@@ -926,6 +954,35 @@ RealtimeEffectPanel::RealtimeEffectPanel(wxWindow* parent, wxWindowID id, const
    mEffectList = effectList;
 
    SetSizerAndFit(vSizer.release());
+
+   mTrackListChanged = TrackList::Get(mProject).Subscribe([this](const TrackListEvent& evt) {
+         auto track = evt.mpTrack.lock();
+         auto waveTrack = dynamic_cast<WaveTrack*>(track.get());
+
+         if (waveTrack == nullptr)
+            return;
+         
+         switch (evt.mType)
+         {
+         case TrackListEvent::TRACK_DATA_CHANGE:
+            if (mCurrentTrack.lock() == track)
+               mTrackTitle->SetLabel(track->GetName());
+            UpdateRealtimeEffectUIData(*waveTrack);
+            break;
+         case TrackListEvent::DELETION:
+            if (evt.mExtra == 0)
+               HideRealtimeUIForTrack(*waveTrack);
```

Comment:
Why is this not done in case of replacements (as indicated by evt.mExtra)?  I think I do understand that this can happen if you do a destructive effect application via other UI, or you finish recording, if I look for uses of `TrackList::Replace`.  The commit comment gave some explanation, and this code should too.

But then in the case of replacement, shouldn't there be another branch to call `UpdateRealtimeEffectUIData`?  The Track object is replaced with another one which is now the current state of the track in the project.